docs/site/Components.md

```diff
@@ -0,0 +1,29 @@
+---
+lang: en
+title: 'Components'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Components.html
+---
+
+## Overview
+
+A `Component` is an extensibility class, which allows for creating various types
+of LoopBack 4 extensions.
+
+Apart from its own properties, `Component` class can have the following
+properties:
+
+- `controllers` - An array of [controller](Controllers.md) classes.
+- `providers` - A map of providers to be bound to the application context.
+- `classes` - A map of TypeScipt classes to be bound to the application context.
+- `servers` - A map of name/class pairs for [servers](Server.md).
+- `lifeCycleObservers` - An array of [life cycle observers](Life-cycle.md).
+- `bindings` - An array of [bindings](Bindings.md) to be aded to the application
+  context.
```

Comment:
For some of these artifact types, we have dedicated pages showing how to build extensions contributing such artifacts. Would it make sense to link to those pages from here?

- https://loopback.io/doc/en/lb4/Creating-decorators.html
- https://loopback.io/doc/en/lb4/Creating-servers.html
- https://loopback.io/doc/en/lb4/Extending-request-body-parsing.html

Also please add decorators, body parsers, repositories and booters to your list.

Reply:
@bajtos that is a list of root level properties a Component can have. Repositories and Booters can be added via mixins, not sure how decorators, body parsers would fit in that list.

Reply:
Let's not worry about decorators and servers for now.

However, I feel it's very important to mention body parsers.

See https://loopback.io/doc/en/lb4/Extending-request-body-parsing.html#contribute-a-body-parser-from-a-component

> A component can add one or more body parsers via its bindings property